rtribbia/question1.js

```diff
@@ -0,0 +1,42 @@
+//----------------------------------
+//  1 A
+//----------------------------------
+
+function Ctor() {
+	this.a = 0;
+	this.b = 1;
+}
+
+var obj1 = new Ctor();
+var obj2 = new Ctor();
+var obj3 = {};
+Ctor.call(obj3);
+
+obj1.c = 2;
+obj1.__proto__.d = 3;
+
+
+1A. Obj 1 & 2 show property 'd' as 3 because they both have the same constructor prototype (w magic stamp)
+	Obj 3 on the other hand, used Ctor as a borrowed initializer (temporary method) so when changes were made to the prototype, only objs 1 2 inherited them
+
+
+
+//----------------------------------
+//  1 B
+//----------------------------------
+
+function A() {};
+//set default values for instances of A:
+A.prototype = {num:0, str:'default'};
+var objA = new A();
+
+function B() {};
+// set default values for instances of B:
+B.prototype.num = 0;
+B.prototype.str = 'default';
+var objB = new B();
+
+1B. no idea? Maybe objA treats it's prototype as an object with only two properties. objB's prototype might not be an object in the same sense, also might have more than those 2 properties?
```

Comment:
Any time you create a function, JavaScript creates a `prototype` object for it just in case you decide to use it as a Constructor. That `prototype` will have a property `constructor` which points back to the original function. With function B, you're adding properties to the automatically generated `prototype`, while with function A, you're _overwriting_ it. this means that the `constructor` property gets reset to the `constructor` of a generic object, which is   `Object`.